scipy/stats/tests/test_stats.py

```diff
@@ -6969,70 +6969,92 @@ def test_dist_perm(self):
         assert_approx_equal(pvalue_dist, 0.001, significant=1)
 
 
-# I do not run the original fortran code, some test case came from
-# https://github.com/nolanbconaway/poisson-etest/blob/master/test_etest.py
-# I already ask permission to use the test cases. The original implementation
-# have limitation for high value of count and nobs, we also test for that
-class TestPoissonETest(object):
-
-    # should return the same value
-    def test_same_results(self):
-        count1, count2 = 20, 20
-        nobs1, nobs2 = 10, 10
-        _, pval = stats.poisson_means_test(count1, nobs1, count2, nobs2)
-        assert_almost_equal(pval, 0.999999756892963, decimal=5)
-
-    # should return different value because this implementation solves
-    # odd behavior for large numbers found in original fortran code
+class TestPoissonETest:
+
+    @pytest.mark.parametrize("c1, n1, c2, n2, p_expect", (
+        # example from [1], 6. Illustrative examples: Example 1
+        [0, 100, 3, 100, 0.0884],
+        [2, 100, 6, 100, 0.1749]
+    ))
+    def test_paper_examples(self, c1, n1, c2, n2, p_expect):
+        res = stats.poisson_means_test(c1, n1, c2, n2)
+        assert_allclose(res.pvalue, p_expect, atol=1e-4)
+
+    @pytest.mark.parametrize("c1, n1, c2, n2, p_expect, alt, d", (
+        # These test cases are produced by the wrapped fortran code from the
+        # original authors. Using a slightly modified version of this fortran,
+        # found here, https://github.com/nolanbconaway/poisson-etest,
+        # I created additional tests.
+        [20, 10, 20, 10, 0.9999997568929630, 'two-sided', 0],
+        [10, 10, 10, 10, 0.9999998403241203, 'two-sided', 0],
+        [10, 10, 10, 10, 0.741441295832156, 'two-sided', .1],
+        [3, 100, 20, 300, 0.12202725450896404, 'two-sided', 0]
+    ))
+    def test_fortran_authors(self, c1, n1, c2, n2, p_expect, alt, d):
+        res = stats.poisson_means_test(c1, n1, c2, n2, alternative=alt, diff=d)
+        assert_allclose(res.pvalue, p_expect, atol=1e-6)
+
     def test_different_results(self):
+        # The implementation in Fortran is known to break down at higher
+        # counts and observations, so we expect different results.
         count1, count2 = 10000, 10000
         nobs1, nobs2 = 10000, 10000
-        _, pval = stats.poisson_means_test(count1, nobs1, count2, nobs2)
+        res = stats.poisson_means_test(count1, nobs1, count2, nobs2)
         with assert_raises(AssertionError):
-            assert_almost_equal(pval, 0.24866994128694545, decimal=5)
+            assert_allclose(res.pvalue, 0.248669941286945455)
```

Comment:
`assert_allclose` is preferred to `assert_almost_equal`. [See NumPy documentation](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_almost_equal.html).